### PR TITLE
Support camel dataformat configuration for marshaller/unmarshaller #816

### PR DESCRIPTION
**Pull request by luigidemasi**
this pr allow the configuration of a dataformat directly in properties file, something like:

```
camel.source.unmarshal=fhirJson
camel.source.unmarshal.fhirJson.fhirVersion=R4
camel.source.unmarshal.fhirJson.prettyPrint=true
```

### CODE REVIEW

**Review by oscerd — APPROVED** [review]
Looks good to me. Thanks a lot. @valdar cc

**Comment by oscerd** [comment]
Can you fix the codestyle?

File: core/src/main/java/org/apache/camel/kafkaconnector/utils/CamelKafkaConnectDataformat.java [hunk 1]
```diff
@@ -35,6 +80,147 @@ public CamelKafkaConnectDataformatKind getDataformatKind() {
...
+
+    public static DataFormatDefinition configureDataformat(CamelContext context, String dataFormatName, Predicate<String> keyFilter, Properties camelProperties) {
+        Map<String, Object> dataFormatProps = camelProperties.stringPropertyNames()
+                .stream()
+                .filter(keyFilter)
+                .collect(Collectors.toMap(key -> key.substring(key.lastIndexOf(".") + 1), key -> camelProperties.get(key)));
+
+        DataFormatDefinition dataformat;
+
+        switch (dataFormatName) {
```
Comment (valdar) [inline core/src/main/java/org/apache/camel/kafkaconnector/utils/CamelKafkaConnectDataformat.java:115]: I hope this switch case could be avoided by something on the line of `context.createDataFormat(dataFormatName)`

**Review by valdar — CHANGES_REQUESTED** [review]
I was under the general impression that this usecase should have worked already...
Give me some more time to investigate further.

**Comment by luigidemasi** [comment]
> I was under the general impression that this usecase should have worked already using something like:
> 
> ```
> camel.dataformat.fhirJson.fhirVersion=R4
> camel.dataformat.fhirJson.prettyPrint=true
> ```
> 
> see https://github.com/apache/camel-kafka-connector/blob/master/core/src/test/java/org/apache/camel/kafkaconnector/DataFormatTest.java#L128
> Give me some more time to investigate further.

It should work but it doesn't, if you look at the test, you disable validation and then assert that the validation is enabled.
I think it's a bug on camel-main atuoconfiguration that doesn't bind the dataformat in the registry and a new one is created when the route starts. Opened an issue for this: [CAMEL-16032](https://issues.apache.org/jira/browse/CAMEL-16032)

**Comment by valdar** [comment]
Thanks a lot for the effort in seeking the root cause of this and fixing it @luigidemasi ! 
I close this and the effort is tracked here: https://github.com/apache/camel-kafka-connector/issues/816

**Pull request closed without merging**